src/main/kotlin/org/rust/ide/refactoring/implementMembers/impl.kt

```diff
@@ -49,6 +55,59 @@ private fun insertNewTraitMembers(selected: Collection<RsAbstractable>, existing
     if (selected.isEmpty()) return
 
     val templateImpl = RsPsiFactory(existingMembers.project).createMembers(selected, trait.subst)
+    val mod = existingMembers.containingMod
+    val superMods = LinkedHashSet(mod.superMods)
+
+    val importCandidates = mutableListOf<ImportCandidate>()
+    val importCandidateVisitor = object : RsVisitor() {
+        override fun visitFunction(o: RsFunction) {
+            o.typeParameterList?.let(this::visitTypeParameterList)
+            o.valueParameterList?.let(this::visitValueParameterList)
+
+            o.whereClause?.wherePredList?.forEach {
+                it.typeReference?.let(this::visitTypeReference)
+            }
+        }
+
+        override fun visitValueParameterList(o: RsValueParameterList) {
+            o.valueParameterList.forEach { it.typeReference?.accept(this) }
+        }
+
+        override fun visitTypeParameterList(o: RsTypeParameterList) {
+            o.typeParameterList.forEach { it.typeReference?.accept(this) }
+        }
+
+        override fun visitRetType(o: RsRetType) {
+            o.typeReference?.let(::visitTypeReference)
+        }
+
+        override fun visitTypeReference(o: RsTypeReference) {
+            val ty = o.type
+            val item: RsQualifiedNamedElement = when (ty) {
+                is TyAdt -> ty.item
+                is TyTraitObject -> ty.trait.element
+                else -> return
+            }
+
+            val candidate = QualifiedNamedItem.ExplicitItem(item)
+                .withModuleReexports(mod.project)
+                .mapNotNull { ImportCandidate(it, it.canBeImported(superMods) ?: return@mapNotNull null) }
+                .firstOrNull() ?: return
+
+            importCandidates.add(candidate)
+        }
+    }
+
+    // Visit the existing trait functions to determine the types that need to be imported.
```

Comment:
Another example:

```rust
// lib.rs
use std::collections::BTreeSet;

pub struct S;

pub trait T {
    fn foo(s: BTreeSet<i32>);
}

// main.rs before using `Implement Members`
use sandbox::T;

struct Q;
impl/*caret*/ T for Q {
}

// main.rs after using `Implement Members`
use sandbox::T;
use alloc::collections::btree_set::set::BTreeSet;

struct Q;
impl T for Q {
    fn foo(s: BTreeSet<i32> /*<- still unresolved*/) {
        unimplemented!()
    }
}
```

Reply:
Nice catch and good reminder for me to update/add some more test-cases to https://github.com/intellij-rust/intellij-rust/blob/master/src/test/kotlin/org/rust/ide/refactoring/implementMembers/ImplementMembersHandlerTest.kt